main.cpp

```diff
@@ -300,6 +300,25 @@ TEST_CASE("Named arguments")
     REQUIRE(fullName == "JamesBond");
 }
 
+TEST_CASE("Named arguments in any order")
+{
+    using FirstName = fluent::NamedType<std::string, struct FirstNameTag>;
+    using LastName = fluent::NamedType<std::string, struct LastNameTag>;
+    static const FirstName::argument firstName;
+    static const LastName::argument lastName;
```

Comment:
I don't see how, but it would be nice to make the usage code shorter.
This is nice and I'll validate the PR, but as we have little feedback on it I'll mark it as experimental in the API.
Thanks for the PR!